src/utils.jl

```diff
@@ -28,42 +28,65 @@ function get_args(func_def::Dict)
   arg_list, kwarg_list, arg_dict
 end
 
+function forward_args(func_def)
```

Comment:
Could you add comments or docstrings in front of this function definition with a why/what explatation of its purpose?